bam_ampliconclip.c

```diff
@@ -609,15 +658,36 @@ static int bam_clip(samFile *in, samFile *out, samFile *reject, char *bedfile,
                     if (is_rev) {
                         if (bam_trim_right(b, b_tmp, p_size, clipping) != 0)
                             goto fail;
+
                         swap_bams(&b, &b_tmp);
                         r_count++;
                     } else {
                         if (bam_trim_left(b, b_tmp, p_size, clipping) != 0)
                             goto fail;
+
                         swap_bams(&b, &b_tmp);
                         f_count++;
                     }
 
+                    if (param->oa_tag) {
+                        uint8_t *tag;
+
+                        if ((tag = bam_aux_get(b, "OA")))
+                            bam_aux_del(b, tag);
+
+                        bam_aux_append(b, "OA", 'Z', strlen(oat.s) + 1, (uint8_t *)oat.s);
```

Comment:
This could be done more easily by calling `bam_aux_update_str(b, "OA", oat.l + 1, oat.s)`.  It can handle both replacing an existing tag and adding a new one.

There should also be a check to ensure it worked.

Reply:
There is a `bam_aux_update_str`?  Excellent, I will use that instead.